fingerprint/infra/scanner/src/main/java/com/simprints/fingerprint/infra/scanner/capture/FingerprintCaptureWrapperV2.kt

```diff
@@ -12,159 +12,94 @@ import com.simprints.fingerprint.infra.scanner.v2.domain.main.message.un20.model
+        scannerV2.captureFingerprint(captureDpi).ensureCaptureResultOkOrError()
+        tracker.completeScan()
+        val qualityScore = scannerV2.getImageQualityScore()
+            ?: throw NoFingerDetectedException("Failed to acquire image quality score")
+        validateMinimumFingerImageQuality(qualityScore)
+
+        // If the quality score is below the threshold and we don't allow low quality extraction, return an empty template
+        if (qualityScore < qualityThreshold && !allowLowQualityExtraction) {
+            AcquireFingerprintTemplateResponse(
+                ByteArray(0), templateFormat, qualityScore
+            )
+
+        } else {
+            acquireTemplateAndAssembleResponse(qualityScore)
+                ?: throw NoFingerDetectedException("Failed to acquire template")
         }
+    }
 
-    private fun Single<Int>.validateMinimumFingerImageQuality() =
-        flatMap { qualityScore ->
-            if (qualityScore > NO_FINGER_IMAGE_QUALITY_THRESHOLD) {
-                Single.just(qualityScore)
-            } else {
-                Single.error(NoFingerDetectedException("Image quality score below detection threshold"))
-            }
-        }
+    private fun CaptureFingerprintResult.ensureCaptureResultOkOrError() = when (this) {
+        CaptureFingerprintResult.OK -> { /* Do nothing */ }
+        CaptureFingerprintResult.FINGERPRINT_NOT_FOUND -> throw NoFingerDetectedException("Fingerprint not found")
+        CaptureFingerprintResult.DPI_UNSUPPORTED -> throw UnexpectedScannerException("Capture fingerprint DPI unsupported")
+        CaptureFingerprintResult.UNKNOWN_ERROR -> throw UnknownScannerIssueException("Unknown error when capturing fingerprint")
+    }
 
-    private fun Single<Int>.acquireTemplateAndAssembleResponse() =
-        flatMapMaybe { imageQuality ->
-            scannerV2.acquireTemplate()
-                .map { templateData ->
-                    AcquireFingerprintTemplateResponse(
-                        templateData.template, templateFormat, imageQuality
-                    )
-                }
+    private fun validateMinimumFingerImageQuality(qualityScore: Int) {
+        if (qualityScore <= NO_FINGER_IMAGE_QUALITY_THRESHOLD) {
+            throw NoFingerDetectedException("Image quality score below detection threshold")
         }
+    }
 
-
-    private fun Single<AcquireFingerprintTemplateResponse>.ifNoFingerDetectedThenSetBadScanLedState() =
-        onErrorResumeNext {
-            if (it is NoFingerDetectedException) {
-                scannerV2.setSmileLedState(scannerUiHelper.badScanLedState())
-                    .andThen(Single.error(it))
-            } else {
-                Single.error(it)
-            }
+    private suspend fun acquireTemplateAndAssembleResponse(imageQuality: Int) =
+        scannerV2.acquireTemplate()?.template?.let { templateData ->
+            AcquireFingerprintTemplateResponse(
+                templateData, templateFormat, imageQuality
```

Comment:
The first parameter used to be `templateData.template` instead of just `templateData`

```
AcquireFingerprintTemplateResponse(
                        templateData.template, templateFormat, imageQuality
                    )
```

Just double checking that the correct param is being passed. Feel free to resolve if all good

Reply:
Yes, I confirm that I am providing the correct parameters.